### PR TITLE
Update the README documentation

### PR DESCRIPTION
**Pull request by amaier17**
Fixed up the README documentation to be proper markdown and updated it to the latest information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# p2pmem-test
+# Introduction
```
Comment (sbates130272) [inline README.md]: @amaier17 can we start with

"# p2pmem-test: A tool for userspace testing of p2pdma

## Introduction"

and then move forward from there?

File: README.md [hunk 2]
```diff
@@ -9,55 +9,41 @@ features.
...
 
 p2pmem-test is an active project. We will happily consider Pull
-Requests (PRs) submitted against the offical repo at
-www.github.com/sbates130272/p2pmem-test.git.
+Requests (PRs) submitted against the [official repo][1].
 
 ## Getting Started
+1. In order to run any P2P traffic, you'll need to have a (Linux based) OS
+that supports [p2pdma][2]. This module is available in all Linux kernels 4.20
+or newer, however it is NOT compiled in by default.
```
Comment (sbates130272) [inline README.md]: Add "So you will almost certainly have to compile a kernel from source and install that. The instructions on how to do that are beyond the scope of this document but there is a tool that can help here (ref to my build-latest-p2pdma tool).

File: README.md [hunk 3]
```diff
@@ -9,55 +9,41 @@ features.
...
-
-0. Make sure your system has at least one p2pmem capable
-device. Examples include an Eideticom IOMEM device, a Microsemi NVRAM 
+1. Make sure your system has at least one p2pdma capable
+device. Examples include an Eideticom IOMEM device, a Microsemi NVRAM
 card or a CMB enabled NVMe SSD that supports the WDS and RDS features
-(e.g. The Everspin NVNitro card or the Eideticom NoLoad
-HDK). Basically this is any PCI EP capable of exposing a BAR with a
-driver that ties into the p2pmem framework.
+(e.g. The Everspin NVNitro card or the Eideticom NoLoad<sup>TM</sup> device).
```
Comment (sbates130272) [inline README.md]: Link NoLoad to our NoLoad product page.

File: README.md [hunk 4]
```diff
@@ -9,55 +9,41 @@ features.
...
-options.
+	Ideally place the two (or more) devices noted in the two previous points behind
+	a PCIe switch (for example the Microsemi Switchtec or a PLX switch). If you do
+	not have a switch you can connect both devices to the Root Complex (RC) on the
+	CPU but two things *may* happen:
+	* Performance may drop. Many RCs are inefficient at routing P2P traffic.
+	* It might not work at all. Many RCs block P2P traffic.
+
+1. You'll almost certainly want to disable the IOMMU in your system
+via both the BIOS and the kernel.
```
Comment (sbates130272) [inline README.md]: s/both/either/ you should only need one.

**Review by sbates130272 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (amaier17) [reply README.md]: I want to add this as a section as well.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# p2pmem-test
+# p2pmem-test: A tool for userspace testing of the p2pdma
```
Comment (sbates130272) [inline README.md]: the p2pdma kernel framework

File: README.md [hunk 6]
```diff
@@ -9,55 +9,44 @@ features.
...
 
 ## Getting Started
-
-0. Make sure the (Linux based) OS supports p2pmem [1]. p2pmem is not
-upstream (yet) so you will need to grab it and build you own kernel.
-
-0. Make sure your system has at least one p2pmem capable
-device. Examples include an Eideticom IOMEM device, a Microsemi NVRAM 
+1. In order to run any P2P traffic, you'll need to have a (Linux based) OS
+that supports [p2pdma][2]. This module is available in all Linux kernels 4.20
```
Comment (sbates130272) [inline README.md]: s/module/framework/

File: README.md [hunk 7]
```diff
@@ -9,55 +9,44 @@ features.
...
-debugsfs entry for each p2pmem device in <debugfs mount
-point>/p2pmem/p2pmemN.
-
-0. Compile p2pmem-test by runnning make. Note p2pmem-test using
-libargconfig as a submodule so make sure you have run submodule init
-to pull that code.
-
-0. You can now run p2pmem-test. Use p2pmem-test -h to get a list of
-options.
+	Ideally place the two (or more) devices noted in the two previous points behind
```
Comment (sbates130272) [inline README.md]: Is this indent intended?

File: README.md [hunk 8]
```diff
@@ -9,55 +9,44 @@ features.
...
+	not have a switch you can connect both devices to the Root Complex (RC) on the
+	CPU but two things *may* happen:
+	* Performance may drop. Many RCs are inefficient at routing P2P traffic.
+	* It might not work at all. Many RCs block P2P traffic.
+
+1. You'll almost certainly want to disable the IOMMU in your system
+via either the BIOS or the kernel.
+
+1. You may also need to disable the PCIe Access Control Services (ACS)
+by either the BIOS and/or the kernel as TLP redirection (activated as part of
```
Comment (sbates130272) [inline README.md]: s/as/because/

File: README.md [hunk 9]
```diff
@@ -19,6 +19,10 @@ have to compile a kernel from source and install that. The instructions on how
 to do this are beyond the scope of this document but there is a tool that can
 help [here][3].
 
+1. You'll need install the separate [p2pmem-pci][5] module installed to expose
```
Comment (sbates130272) [inline README.md:22]: Can we just squash this into the other commit?

**Review by sbates130272 — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (amaier17) [reply README.md]: Yes. This is so that it shows up as part of the number above it.

Comment (amaier17) [reply README.md]: I've updated it to be a part of the previous block.

**Review by sbates130272 — APPROVED** [review]
Looks great. Thanks @amaier17!

**Pull request merged**